expression/builtin_time.go

```diff
@@ -1388,31 +1389,49 @@ func (c *currentTimeFunctionClass) getFunction(args []Expression, ctx context.Co
 	if err := c.verifyArgs(args); err != nil {
 		return nil, errors.Trace(err)
 	}
-	sig := &builtinCurrentTimeSig{newBaseBuiltinFunc(args, ctx)}
+
+	if len(args) == 0 {
+		bf := newBaseBuiltinFuncWithTp(args, ctx, tpDuration)
+		bf.tp.Flen, bf.tp.Decimal, bf.deterministic = mysql.MaxDurationWidthNoFsp, types.MinFsp, false
+		sig := &builtinCurrentTime0ArgSig{baseDurationBuiltinFunc{bf}}
+		return sig.setSelf(sig), nil
+	}
+	// args[0] must be an in constant which will not be null.
```

Comment:
if you make assumptions that `arg[0]` is a `Constant`, you should do the type assertion before calling `args[0].EvalInt`

Reply:
Here args[0] can only be an int constant for current_time which is promised be parser.

Reply:
I see. But we still need to check the `NULL` value here. Once parser changes its behavior and not guarantee the parameter to be a constant or maybe parser passes a NULL constant, we can detect this change and make the right move.